src/qubx/impl/ccxt_connector.py

```diff
@@ -117,28 +127,30 @@ def subscribe(
 
         return True
 
-    def get_communication_channel(self) -> CtrlChannel:
-        return self._ch_market_data
-
-    def _check_existing_subscription(self, subscription_type, symbols: List[str]) -> List[str]:
+    def _check_existing_subscription(self, subscription_type, instruments: List[Instrument]) -> List[str]:
         subscribed = self._subsriptions[subscription_type]
         to_subscribe = []
-        for s in symbols:
-            if s not in subscribed:
-                to_subscribe.append(s)
+        for s in instruments:
+            if s.symbol not in subscribed:
+                to_subscribe.append(s.symbol)
         return to_subscribe
 
     def _time_msec_nbars_back(self, timeframe: str, nbarsback: int) -> int:
-        return (self.time() - nbarsback * pd.Timedelta(timeframe)).asm8.item() // 1000000
+        return (self.trading_service.time() - nbarsback * pd.Timedelta(timeframe)).asm8.item() // 1000000
```

Comment:
@dmarienko I understood the issue in paper trading! self.trading_service.time() returns "1970-01-01T00:00:00.000000000", so when we calculate the `since` parameter for querying candles, it is completely incorrect. So we have to get time somehow differently

Reply:
```suggestion
        return (self.time() - nbarsback * pd.Timedelta(timeframe)).asm8.item() // 1000000
```